src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs

```diff
@@ -224,12 +256,17 @@ public void WriteLine(VerbosityLevel verbosity, string format, params object[] a
         {
             ThreadHelper.ThrowIfNotOnUIThread();
 
-            if (OutputVerbosity >= (int)verbosity && _outputConsole != null)
+            if (ShowMessageAsOutput(verbosity))
             {
                 _outputConsole.WriteLine(format, args);
             }
         }
 
+        public bool ShowMessageAsOutput(VerbosityLevel verbosity)
```

Comment:
Suggestion: `ShowMessageAsOutput` implies action would be taken. `ShouldShowMessageAsOutput` would be more precise.